### DISCUSSION TITLE
[v4] Unnecessary Variables Causing Build Size Increase

### DISCUSSION THREAD

**Discussion by Harshkt214** [tailwindlabs/tailwindcss#15150] [General]
I'm encountering an issue with @theme{}. Even though I'm not adding any classes to my project, it's still including unused design tokens in the build, which is increasing the size by approximately 17KB for just the base styles. I've tried removing all design tokens, but this process is time-consuming [check here](https://tailwindcss.com/docs/v4-beta#overriding-the-default-theme).

Check out this repo [Harshkt214/material-theme](https://github.com/Harshkt214/material-theme).
use `bun build` to create production build.

**Comment by wongjn** [comment]
I believe this is intended at the moment, [as per the docs](https://tailwindcss.com/docs/v4-beta#css-theme-variables):

> Tailwind CSS v4.0 takes all of your design tokens and makes them available as CSS variables by default, so you can reference any value you need at run-time using just CSS.

**Reply by Harshkt214** [reply]
I think they must have to add automatic remove of unused styles at build time like before, because removing design tokens manually causes error in development time.

What you think? I report it as a bug.
